src/containers/sound-editor.jsx

```diff
@@ -209,11 +202,13 @@ class SoundEditor extends React.Component {
             newSamples.set(firstPart, 0);
             newSamples.set(secondPart, firstPart.length);
         }
-        this.submitNewSamples(newSamples, sampleRate);
-        this.setState({
-            trimStart: null,
-            trimEnd: null
+        this.submitNewSamples(newSamples, sampleRate).then(() => {
+            this.setState({
+                trimStart: null,
+                trimEnd: null
+            });
         });
+
     }
     handleDeleteInverse () {
```

Comment:
I know this is outside the scope of your change, but I'm confused about what "handleDeleteInverse" means

Reply:
deleteInverse is one of the operations you can take on a sound selection. It deletes everything you do not have selected (the inverse of the selection). You do this by shift+delete, so it is a bit of a hidden feature.

Reply:
Ah, that makes sense. Could it make sense to add a comment above this function saying something like "deletes everything you do not have selected (the inverse of the selection)" ?

Reply:
FWIW I think it's better to put a comment that describes the whole function above the function (or on the same line as its declaration)-- below the declaration line, it sorta looks like it just refers to the variable declarations.